slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -93,18 +133,24 @@ def $name($args): $name = {
 
     trait PlainSqlMapperDef extends super.PlainSqlMapperDef{
       def code = {
-        val positional = compoundValue(columnsPositional.map(c => (if(c.fakeNullable || c.model.nullable)s"<<?[${c.rawType}]"else s"<<[${c.rawType}]")))
+        def mkArgs(values: Seq[String] ) = {
+          if (values.size == 1) s"""( ${values.head} )"""
+          else values.mkString("(" ," , " , " )")
+        }
+
+        val positional = mkArgs(columnsPositional.map(c => (if(c.fakeNullable || c.model.nullable)s"<<?[${c.rawType}]"else s"<<[${c.rawType}]")))
         val dependencies = columns.map(_.exposedType).distinct.zipWithIndex.map{ case (t,i) => s"""e$i: GR[$t]"""}.mkString(", ")
-        val rearranged = compoundValue(desiredColumnOrder.map(i => if(hlistEnabled) s"r($i)" else tuple(i)))
-        def result(args: String) = if(mappingEnabled) s"$factory($args)" else args
+        val rearranged = mkArgs(desiredColumnOrder.map(i => tuple(i)) )
+        def result(args: String) = s"""${TableClass.elementType}${args}""" 
         val body =
-          if(autoIncLastAsOption && columns.size > 1){
+          if(autoIncLastAsOption && columns.size > 1 && columns.size <= 22 ){
             s"""
 val r = $positional
```

Comment:
this now breaks for tables > 22. we need to use tupleBuilder here and match below